core-web/apps/dotcms-ui/src/app/portlets/shared/directives/dot-dropdown.directive.ts

```diff
@@ -42,14 +48,18 @@ export class DotDropdownDirective {
     @Input()
     set dotOptions(options: Array<DotDropdownSelectOption<string>>) {
         if (options) {
-            this.primeDropdown.options = options.map((opt) => {
-                return {
-                    ...opt,
-                    [DEFAULT_LABEL_NAME_INDEX]: this.dotMessagePipe.transform(
-                        opt[DEFAULT_LABEL_NAME_INDEX]
-                    )
-                };
-            });
+            this.setOptions(options);
         }
     }
+
+    private setOptions(options: Array<DotDropdownSelectOption<string>>) {
+        this.control.options = options.map((opt) => {
+            return {
+                ...opt,
+                [DEFAULT_LABEL_NAME_INDEX]: this.dotMessagePipe.transform(
```

Comment:
Don't know what to think about the fact that we have a pipe we're calling directly the transform method:

`this.dotMessagePipe.transform`

Reply:
Moved to use the service directly instead of the pipe. Removed the intermediary.